docs/tools/editor/sublime3.md

```diff
@@ -0,0 +1,248 @@
+| `Ctrl+Shift+k`              | 删除行                                                       |
+| `Ctrl+Enter`                | 在下方插入行                                                 |
+| `Ctrl+Shitf+Enter`          | 在上方插入行                                                 |
+| `Ctrl+Shitf+Up`             | 行上移                                                       |
+| `Ctrl+Shitf+Down`           | 行下移                                                       |
+| `Ctrl+L`                    | 选择行，重复以向下选择多行                                   |
+| `Ctrl+D`                    | 选择词，重复以选择多个相同词，并进入多点编辑（用于快速批量更改） |
+| `Ctrl+M`                    | 跳转到匹配的括号                                             |
+| `Ctrl+Shift+M`              | 选择括号内的内容，重复以包括括号                             |
+| `Ctrl+K` & ` Ctrl+K`        | 删至行尾（复合快捷键，建议使用 vim 模式代替）                |
+| `Ctrl+K` & `Ctrl+Backspace` | 删至行首（复合快捷键，建议使用 vim 模式代替）                |
+| `Ctrl+]`                    | 缩进当前（选择的）行                                         |
+| `Ctrl+[`                    | 取消缩进当前（选择的）行                                     |
+| `Ctrl+Shift+D`              | 复制当前行，并插入在下一行                                   |
+| `Ctrl+J`                    | 合并下一行与当前行                                           |
+| `Ctrl+Shift+V`              | 粘贴并缩进（用于整段粘贴代码）                               |
+| `Ctrl+K` & `Ctrl+Shift+V`   | 从历史粘贴（复合快捷键，建议修改为 `Ctrl+Alt+V `）           |
+| `Ctrl ␣`                    | 选择下一个自动补全建议（与 <kbd>Down</kbd> 等价，注意是空格） |
+| `Crtl+Alt+Down`             | 光标下移，并保留当前行光标（进入多点编辑模式）               |
+| `Crtl+Alt+Up`               | 光标上移，并保留当前行光标（进入多点编辑模式）               |
+| ~                           | 转换选择内容的大小写                                         |
+
+### 自动补全
+
+ST3 有丰富的补全功能，可能的补全内容会在光标下方显示，按 <kbd>Tab</kbd> 进行补全。
+
+补全后一般会选择一块区域进行编辑，编辑完成后再次按 <kbd>Tab</kbd> 进行正常代码的编辑（ for 循环等复杂代码可能有多个编辑块，需要按多次 <kbd>Tab</kbd> ）。
+
+自动补全的风格无法调整，除非使用插件。
+
+如果没有自动补全，请如下修复：
+
+1. 检查是否切换了语言， ST3 默认新建文件为 Plain Text。
+
+2. 进入 `首选项->设置` 然后添上两行：
+
+   ```json
+   "auto_complete": true,
+   "auto_match_enabled": true,
+   ```
+
+   
+
+## 编译与运行
+
+ST3 的编译环境已经配置好了，可以直接使用。
+
+Windows 环境下需要将 g++ 所在目录添加到环境变量中，并将其优先级调至最高。
+
+### 编译
+
+直接按 `Ctrl+B` 编译，第一次使用会让你选择编译方式，选择 `C++ Single File ` 或 `C++ Single File - Run` 。
+
+#### 修改编译选项
+
+虽然 ST3 自带编译环境，但不幸的是，编译选项无法修改。
+
+如果需要修改编译选项，则需要自己新建一个编译系统。
+
+进入 `系统->编译系统->新建编译系统…` 然后在大括号中间输入：
+
+```json
+ 	"cmd": ["cmd","/c","g++","-Wall","-lm","${file}","-o","${file_path}/${file_base_name}","&&","cmd","/c","${file_path}/${file_base_name}"],    //编译选项可以自己调整
+	"file_regex": "^(..[^:]*):([0-9]+):?([0-9]+)?:? (.*)$",    //这一行可以让ST3图形化显示报错，如果习惯了看g++返回的信息可以去掉
+```
+
+保存后把编译系统切换到你自己写的这个，就可以使用了，这里的配置是在ST3窗口内编译并运行。
+
+保存的文件为 `C:\Users\用户名\AppData\Roaming\Sublime Text 3\Packages\User\编译系统名.sublime-build` 可以反复修改。
+
+??? note "其他配置方法"
+    以下为满足各位的各种需要而写，可能需要掌握一些命令行的知识。
+
+    只编译 
+    
+    ```json
+        "cmd": [“g++","-Wall","-lm","${file}","-o","${file_path}/${file_base_name}"],

```

Comment:
```suggestion
        "cmd": [“g++","-Wall","${file}","-o","${file_path}/${file_base_name}"],

```
No need `-lm`, it will be automatically added in g++
Ref:
[1]: https://github.com/OI-wiki/OI-wiki/pull/2615#discussion_r479460852
[2]: https://stackoverflow.com/questions/1033898/why-do-you-have-to-link-the-math-library-in-c